src/java.base/share/classes/java/nio/file/spi/package-info.java

```diff
@@ -31,7 +31,10 @@
  *
  * <p> Unless otherwise noted, passing a {@code null} argument to a constructor
  * or method in any class or interface in this package will cause a {@link
- * java.lang.NullPointerException NullPointerException} to be thrown.
+ * java.lang.NullPointerException NullPointerException} to be thrown.  In some
+ * cases methods which are specified to throw an {@code IOException} may throw
+ * a more precise <a href="../file/package-summary.html#optspecex">optional
```

Comment:
The terminology in the reference section is "optional specific exception" so it would be more consistent to continue with "specific" rather than switching to "precise" here. I'm in two minds about adding this to the java.nio.channels package description as it's only applicable to the static open methods defined by FileChannel and AsynchronousFileChannel.

Reply:
Changed "precise" to "specific" and removed update to `java.nio.channels` package doc.